snapcraft/internal/states/_state.py

```diff
@@ -81,6 +86,11 @@ def _get_differing_keys(dict1, dict2):
     return differing_keys
 
 
+def get_global_state():
+    with open(os.path.join('snap', '.snapcraft', 'state'), 'r') as state_file:
```

Comment:
No, and yes. Good catches, but agh, because they are not real lifecycle steps I can't find a good place for their unit tests. I will put them in lifecycle.